### PR TITLE
chore: Add stetho 

### LINKED ISSUE #2368: Add Stetho debug bridge
Docs: https://github.com/facebook/stetho

**Would you like to work on the issue?**

Good first issue

### PR DESCRIPTION
**Pull request by KKS161994**
Added stetho as dependency and initialised it in MainApplication.

### CODE REVIEW

**Review by iamareebjamal — CHANGES_REQUESTED** [review]
Add OkHTTP interceptor and all other integrations

**Comment by KKS161994** [comment]
Can you please let me know if their are some other changes that I need to do.

File: app/src/main/java/org/fossasia/susi/ai/MainApplication.kt [hunk 1]
```diff
@@ -50,6 +53,8 @@ class MainApplication : Application() {
             // Release mode
             Timber.plant(ReleaseLogTree())
         }
+        Stetho.initializeWithDefaults(this)
+        OkHttpClient.Builder().addNetworkInterceptor(StethoInterceptor()).build()
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/MainApplication.kt]: Won't work. Please ensure it works before sending PR

**Comment by KKS161994** [comment]
Sir I have added NetworkInterceptor, Js module, dumpapp, local storage module of stetho. Is their anything more to add sir.
I have tested it from terminal using ./dumpapp and also using inspector of chrome for network requests and local storage.

File: app/build.gradle [hunk 2]
```diff
@@ -181,4 +181,14 @@ dependencies {
     //Smart LOck authentication
     implementation "com.google.android.gms:play-services-auth:${rootConfiguration.playServiceAuthVersion}"
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:${rootConfiguration.kotlinVersion}"
+
+    //Stetho
+    implementation 'com.facebook.stetho:stetho:1.5.1'
+    //Stetho network helper
+    implementation 'com.facebook.stetho:stetho-okhttp3:1.5.1'
+    //Stetho js module
+    implementation 'com.facebook.stetho:stetho-js-rhino:1.4.2'
```
Comment (iamareebjamal) [inline app/build.gradle]: Should only be enabled in debug build

**Comment by KKS161994** [comment]
Sir I have added it as a debug implementation and also the no-op part for stetho so that it doesn't fail for release build too.
releaseCompile "net.igenius:stetho-no-op:1.1"
Above is no-op library for stetho .

File: app/src/main/java/org/fossasia/susi/ai/MainApplication.kt [hunk 3]
```diff
@@ -50,6 +51,12 @@ class MainApplication : Application() {
             // Release mode
             Timber.plant(ReleaseLogTree())
         }
+        // Initializes stetho for dumpapp, database, local storage
+        Stetho.initialize(
+                Stetho.newInitializerBuilder(this)
+                        .enableDumpapp(Stetho.defaultDumperPluginsProvider(this))
+                        .enableWebKitInspector(Stetho.defaultInspectorModulesProvider(this))
+                        .build())
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/MainApplication.kt]: Use initializeWithDefaults

**Comment by KKS161994** [comment]
Sir to use Stetho.initialisewithdefaults() I have to use a different no-op library since the older one didn't have initialiseWithdefaults() as a  member method. The new library has it included with other methods too.

**Comment by iamareebjamal** [comment]
https://github.com/iamareebjamal/stetho-noop

<blockquote><div> GitHub</div><div><strong><a href="https://github.com/iamareebjamal/stetho-noop">iamareebjamal/stetho-noop</a></strong></div><div>Facebook stetho no-op for release build. Contribute to iamareebjamal/stetho-noop development by creating an account on GitHub.</div></blockquote>

**Comment by KKS161994** [comment]
> https://github.com/iamareebjamal/stetho-noop
> 
> >  GitHub**[iamareebjamal/stetho-noop](https://github.com/iamareebjamal/stetho-noop)**Facebook stetho no-op for release build. Contribute to iamareebjamal/stetho-noop development by creating an account on GitHub.

Sir I integrated it for release implementation .

<blockquote><div> GitHub</div><div><strong><a href="https://github.com/iamareebjamal/stetho-noop">iamareebjamal/stetho-noop</a></strong></div><div>Facebook stetho no-op for release build. Contribute to iamareebjamal/stetho-noop development by creating an account on GitHub.</div></blockquote>

File: app/build.gradle [hunk 4]
```diff
@@ -181,4 +181,17 @@ dependencies {
...
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:${rootConfiguration.kotlinVersion}"
+
+    //Stetho
+    debugImplementation "com.facebook.stetho:stetho:${rootConfiguration.stethoVersion}"
+
+    //Stetho network helper
+    debugImplementation "com.facebook.stetho:stetho-okhttp3:${rootConfiguration.stethoVersion}"
+
+    //Stetho js module
+    debugImplementation "com.facebook.stetho:stetho-js-rhino:${rootConfiguration.stehtoJsRhinoVersion}"
```
Comment (iamareebjamal) [inline app/build.gradle]: This isn't required

File: app/config.gradle [hunk 5]
```diff
@@ -25,4 +25,7 @@ ext {
     mockitoCoreVersion = '1.10.19'
     playServiceAuthVersion='16.0.1'
     shimmerVersion = "0.5.0"
+    jitpackVersion = "1.1"
```
Comment (iamareebjamal) [inline app/config.gradle]: There is no library named 'jitpack'

Also, this section id only for dependencies with duplicate versions

**Comment by KKS161994** [comment]
Sir I have removed the js module of Stetho. And also the version numbers that I added. Okhttp dependency is a must to have for monitoring network requests made.

**Review by iamareebjamal — APPROVED** [review]

File: app/build.gradle [hunk 6]
```diff
@@ -181,4 +181,12 @@ dependencies {
     //Smart LOck authentication
     implementation "com.google.android.gms:play-services-auth:${rootConfiguration.playServiceAuthVersion}"
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:${rootConfiguration.kotlinVersion}"
+
+    //Stetho
+    debugImplementation "com.facebook.stetho:stetho:1.5.1"
+
+    //Stetho network helper
+    debugImplementation "com.facebook.stetho:stetho-okhttp3:1.5.1"
```
Comment (iamareebjamal) [inline app/build.gradle]: This version is duplicate. Add it to the versions

**Review by iamareebjamal — CHANGES_REQUESTED** [review]

**Comment by KKS161994** [comment]
Created a version name in the versions for Stetho and stetho ok http sir.

**Review by iamareebjamal — APPROVED** [review]

**Pull request merged**